cypress/e2e/tests/administration/custom-migration-targets/crud.test.ts

```diff
@@ -224,4 +224,52 @@ describe(["@tier1", "@dc", "@interop"], "Custom Migration Targets CRUD operation
             application.delete();
         });
     });
+
+    Object.values(Languages).forEach((language) => {
+        it(`${language} | custom rule with source technology`, function () {
+            const targetData = this.customMigrationTargets["rules_with_source_element"];
+            const target = new CustomMigrationTarget(
+                data.getRandomWord(8),
+                data.getDescription(),
+                targetData.image,
+                getRulesData(targetData),
+                language,
+                targetData.sources
+            );
+            target.create();
+
+            const application = new Analysis(
+                getRandomApplicationData("bookserverApp", {
+                    sourceData: this.appData["bookserver-app"],
```

Comment:
Why do we need to create an analysis object here ? 
We only need an application right?

```
const application = new Application(getRandomApplicationData("bookserverApp", {
                    sourceData: this.appData["bookserver-app"]));  ```

Reply:
We need analysis object, first here we need to select Analysis mode>> Source (we could have also used upload), Then Set target is the Custom target and upload custom rule.